code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -318,10 +319,28 @@ private void track(final Event event) {
         }
 
         // Todo:- Should we append default lifecycle context data (os version, device name, device version, etc) to each hits?
-        final Map<String, Object> contextData = event.getEventData();
-        if(!contextData.isEmpty()) {
-            for (Map.Entry<String, Object> entry : contextData.entrySet()) {
-                processedData.put(entry.getKey(), entry.getValue().toString());
+        final Map<String, Object> dataToProcess = event.getEventData();
```

Comment:
We only have to read and copy the contents of contextData and not the entire eventData.

Reply:
https://github.com/adobe/aepsdk-analyticsedge-ios/blob/main/AEPAnalytics/Sources/Analytics.swift#L130